### PR TITLE
Frustum and planes

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

File: include/nbl/scene/SViewFrustum.h [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+		}
+
+		inline void setFrom(const core::matrix4SIMD& mat)
+		{
+			// not sure if I could use 
+			// inline void setPlane(const vector3df_SIMD& point, const vector3df_SIMD& nvector)
+
+			const auto* data = mat.pointer();
+			auto test = mat[15] + mat[12];
+			// left clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:63]: probably `setPlane`

File: include/nbl/scene/SViewFrustum.h [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				core::vector3df_SIMD normal(data[3] + data[0], data[7] + data[4], data[11] + data[8]);
+				planes[VF_LEFT_PLANE].setNormal(normal);
+				planes[VF_LEFT_PLANE].D = mat[15] + mat[12]; // ? TODO
+			}
+
+			// right clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:70]: probably `setPlane`

File: include/nbl/scene/SViewFrustum.h [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				core::vector3df_SIMD normal(data[3] - data[0], data[7] - data[4], data[11] - data[8]);
+				planes[VF_RIGHT_PLANE].setNormal(normal);
+				planes[VF_RIGHT_PLANE].D = mat[15] - mat[12]; // ? TODO
+			}
+			
+			// top clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:77]: probably `setPlane`

File: include/nbl/scene/SViewFrustum.h [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				core::vector3df_SIMD normal(data[3] - data[1], data[7] - data[5], data[11] - data[9]);
+				planes[VF_TOP_PLANE].setNormal(normal);
+				planes[VF_TOP_PLANE].D = mat[15] - mat[13];
+			}
+		
+			// bottom clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:84]: probably `setPlane`

File: include/nbl/scene/SViewFrustum.h [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				core::vector3df_SIMD normal(data[3] - data[1], data[7] - data[5], data[11] - data[9]);
+				planes[VF_BOTTOM_PLANE].setNormal(normal);
+				planes[VF_BOTTOM_PLANE].D = mat[15] + mat[13];
+			}
+			
+			// far clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:91]: probably `setPlane`

File: include/nbl/scene/SViewFrustum.h [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				core::vector3df_SIMD normal(data[3] - data[2], data[7] - data[6], data[11] - data[10]);
+				planes[VF_FAR_PLANE].setNormal(normal);
+				planes[VF_FAR_PLANE].D = mat[15] - mat[14];
+			}
+			
+			// near clipping plane
```
Comment (AnastaZIuk) [inline include/nbl/scene/SViewFrustum.h:98]: probably `setPlane`

File: include/nbl/core/math/plane3dSIMD.h [hunk 7]
```diff
@@ -83,6 +93,29 @@ class plane3dSIMDf : private vectorSIMDf
...
+			ISREL3D_PLANAR if the point is within the plane. 
+		*/
+		EIntersectionRelation3D classifyPointRelation(const vector3df& point) const
+		{
+			static constexpr float ROUNDING_ERROR_f32 = 0.000001f;
+
+			const auto& normal = getNormal();
+			const float d = dot(vector3df(normal.x, normal.y, normal.z), point).X + w;
+
+			if (d < -ROUNDING_ERROR_f32)
```
Comment (AnastaZIuk) [inline include/nbl/core/math/plane3dSIMD.h:110]: I could not find any rounding define, do we have this one somewhere?

File: include/nbl/core/math/plane3dSIMD.h [hunk 8]
```diff
@@ -83,6 +93,29 @@ class plane3dSIMDf : private vectorSIMDf
...
+				return ISREL3D_BACK;
+
+			if (d > ROUNDING_ERROR_f32)
```
Comment (AnastaZIuk) [inline include/nbl/core/math/plane3dSIMD.h:113]: I could not find any rounding define, do we have this one somewhere?

**Comment by devshgraphicsprogramming** [comment]
You shouldn't need to add these back for anything

**Comment by AnastaZIuk** [comment]
closing then

**Pull request closed without merging**